src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php

```diff
@@ -95,21 +100,68 @@ public function parse(array $item) : array
      *
      * @return array
      */
-    private function parseClass(ResourceMetadata $resourceMetadata, string $resourceClass, string $io, array $visited = []) : array
+    private function parseResource(ResourceMetadata $resourceMetadata, string $resourceClass, string $io, array $visited = []) : array
     {
         $visited[] = $resourceClass;
 
-        $attributes = $resourceMetadata->getAttributes();
         $options = [];
-        $data = [];
 
-        if (isset($attributes['normalization_context']['groups'])) {
-            $options['serializer_groups'] = $attributes['normalization_context']['groups'];
-        }
+        $options['serializer_groups'] = $resourceMetadata->getAttribute(
```

Comment:
IMO the problem is here, shoud should keep the `if(isset())`/`else` structure or both read and write groups are added.